core/services/ocr2/delegate.go

```diff
@@ -589,6 +598,39 @@ type connProvider interface {
 	ClientConn() grpc.ClientConnInterface
 }
 
+func (d *Delegate) newServicesVaultPlugin(
+	ctx context.Context,
+	lggr logger.SugaredLogger,
+	jb job.Job,
+	wrapper *gatewayconnector.ServiceWrapper,
+) (srvs []job.ServiceCtx, err error) {
+	spec := jb.OCR2OracleSpec
+
+	cfg := &vault.Config{}
+	err = json.Unmarshal(spec.PluginConfig.Bytes(), cfg)
+	if err != nil {
+		return nil, fmt.Errorf("failed to instantiate vault plugin: failed to unmarshal plugin config: %w", err)
+	}
+
+	gwconnector := wrapper.GetGatewayConnector()
```

Comment:
This will work just fine with @jinhoonbang 's PR that turns Connector into a gRPC service, right?

We will simply pass around this interface: https://github.com/smartcontractkit/chainlink-common/pull/1233/files#diff-1c433b3619c7830d49f3fc6dabd330e0ef0765d5bf252ebd911c6aec9adfdc06R10